src/runtime/pkg/resourcecontrol/cgroups.go

```diff
@@ -79,7 +80,9 @@ func sandboxDevices() []specs.LinuxDeviceCgroup {
 	for _, device := range defaultDevices {
 		ldevice, err := DeviceToLinuxDevice(device)
 		if err != nil {
-			controllerLogger.WithField("source", "cgroups").Warnf("Could not add %s to the devices cgroup", device)
+			if device != "/dev/mshv" || err != unix.ENOENT{
```

Comment:
We shouldn't hardcode /dev/mshv here, because when you actually use /dev/mshv instead of /dev/kvm, you get the same warning about /dev/kvm.

I think we should rather only warn if neither /dev/kvm or /dev/mshv is present. We'll probably have to refactor this loop a bit, maybe introduce a new list that only contains these two hypervisors.

Reply:
I agree with @sprt, 100% here.

And if none of those are found, we simply abort as we know it's not going to work.

Also, I'd like to still get an INFO message saying that we're successfully using one or another.

Reply:
Hi @sprt and @fidencio!

Thanks for the comments!  I hope I've managed to refactor as you thought.